File: QuotesforMindandSoul/app/src/main/java/ro/danserboi/quotesformindandsoul/ShareButtonOnClickListener.java
package ro.danserboi.quotesformindandsoul;

import android.content.Context;
import androidx.core.app.ShareCompat;
import android.view.View;
import android.widget.TextView;

public class ShareButtonOnClickListener implements View.OnClickListener{
    private final Context context;
    private final TextView mWordsTextView;
    private final TextView mAuthorTextView;

    public ShareButtonOnClickListener(Context context, TextView mWordsTextView, TextView mAuthorTextView) {
        this.context = context;
        this.mWordsTextView = mWordsTextView;
        this.mAuthorTextView = mAuthorTextView;
    }

    @Override
    public void onClick(View view) {
        Utils.buttonAnimation(view);
        String txt = "\"" + mWordsTextView.getText() + "\" - " + mAuthorTextView.getText();
        String mimeType = "text/plain";
        new ShareCompat.IntentBuilder(context)
                .setType(mimeType)
                .setChooserTitle(R.string.share_chooser_title)
                .setText(txt)
                .setSubject(context.getString(R.string.share_subject))
                .startChooser();
    }
}

File: QuotesforMindandSoul/app/src/main/java/ro/danserboi/quotesformindandsoul/ShareButtonOnClickListener.kt
package ro.danserboi.quotesformindandsoul

import android.content.Context
import android.view.View
import android.widget.TextView
import androidx.core.app.ShareCompat

class ShareButtonOnClickListener(private val context: Context, private val mWordsTextView: TextView, private val mAuthorTextView: TextView) : View.OnClickListener {
    override fun onClick(view: View) {
        Utils.buttonAnimation(view)
        val txt: String = "\"" + mWordsTextView.text + "\" - " + mAuthorTextView.text
        val mimeType = "text/plain"
        ShareCompat.IntentBuilder(context)
                .setType(mimeType)
                .setChooserTitle(R.string.share_chooser_title)
                .setText(txt)
                .setSubject(context.getString(R.string.share_subject))
                .startChooser()
    }
}